storage/engine/rocksdb/db.cc

```diff
@@ -39,33 +39,84 @@
 extern "C" {
 #include "_cgo_export.h"
 
-struct DBBatch {
+struct DBEngine {
+  rocksdb::DB* const rep;
+
+  DBEngine(rocksdb::DB* r)
+      : rep(r) {
+  }
+  virtual ~DBEngine() { }
+
+  virtual DBStatus Put(DBSlice key, DBSlice value);
+  virtual DBStatus Merge(DBSlice key, DBSlice value);
+  virtual DBStatus Delete(DBSlice key);
+  virtual DBStatus WriteBatch();
+  virtual DBStatus Get(DBSlice key, DBString* value);
+  virtual DBIterator* NewIter();
+};
+
+struct DBImpl : public DBEngine {
+  rocksdb::Env* const memenv;
+
+  DBImpl(rocksdb::DB* r, rocksdb::Env* m)
```

Comment:
The second argument could use a comment (it's not just the DB's `Env`; it's only set if it is a memory environment that needs to be deleted separately).